server/src/main/java/org/opensearch/search/startree/filter/RangeMatchDimFilter.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.startree.filter;
+
+import org.opensearch.index.compositeindex.datacube.startree.index.StarTreeValues;
+import org.opensearch.index.compositeindex.datacube.startree.node.StarTreeNode;
+import org.opensearch.search.internal.SearchContext;
+import org.opensearch.search.startree.StarTreeNodeCollector;
+import org.opensearch.search.startree.filter.provider.DimensionFilterMapper;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.Optional;
+
+public class RangeMatchDimFilter implements DimensionFilter {
+
+    private final String dimensionName;
+
+    private final Object low;
+    private final Object high;
+    private final boolean includeLow;
+    private final boolean includeHigh;
+
+    private Long lowOrdinal;
+    private Long highOrdinal;
+
+    private boolean skipRangeCollection = false;
```

Comment:
Can we add comments on what this does? 

Also java docs we will need to add for all new classes - which you already might be doing

Reply:
skipCollectionForOutOfBounds maybe ?